Packs/Kafka/Integrations/KafkaV3/KafkaV3.py

```diff
@@ -0,0 +1,729 @@
+        else:
+            raise DemistoException(f'Offset {offset} is not in supported format.')
+
+    if partitions:
+        topic_metadata = topics[topic]
+        available_partitions = topic_metadata.partitions.values()
+        available_partitions_ids = [available_partition.id for available_partition in available_partitions]
+        for partition in partitions:
+            if int(partition) not in available_partitions_ids:
+                raise DemistoException(f"Partition {partition} is not assigned to kafka topic {topic} available "
+                                       f"{available_partitions_ids}.")
+            if check_offset:
+                earliest_offset, oldest_offset = kafka.get_partition_offsets(topic=topic, partition=int(partition))
+                if offset < int(earliest_offset) or offset >= int(oldest_offset):  # type: ignore
+                    raise DemistoException(f'Offset {offset} for topic {topic} and partition {partition} '
+                                           f'is out of bounds [{earliest_offset}, {oldest_offset})')
+
+    return True
+
+
+def fetch_incidents(kafka: KafkaCommunicator, demisto_params: dict) -> None:
+    """Fetch incidents as kafka messages from a specific topic.
+
+    Args:
+        kafka: initialized KafkaCommunicator object to preform actions with.
+        demisto_params: The demisto parameters.
+    """
+    topic = demisto_params.get('topic', '')
+    partitions = handle_empty(argToList(demisto_params.get('partition', '')), [])
+    offset = handle_empty(demisto_params.get('first_fetch', 'earliest'), 'earliest')
+    message_max_bytes = int(handle_empty(demisto_params.get("max_bytes_per_message", 1048576), 1048576))
+    max_messages = int(handle_empty(demisto_params.get('max_fetch', 50), 50))
+    last_fetched_offsets = demisto.getLastRun().get('last_fetched_offsets', {})
+    last_topic = demisto.getLastRun().get('last_topic', '')
+
+    demisto.debug(f"Starting fetch incidents with:\n last_topic: {last_topic}, "
+                  f"last_fetched_offsets: {last_fetched_offsets}, "
+                  f"topic: {topic}, partitions: {partitions}, offset: {offset}, "
+                  f"message_max_bytes: {message_max_bytes}, max_messages: {max_messages}\n")
+    incidents = []
+
+    kafka.update_conf_for_fetch(message_max_bytes=message_max_bytes)
+
+    kafka_consumer = KConsumer(kafka.conf_consumer)
+    check_params(kafka, topic, partitions, offset, True)
+
+    if topic != last_topic:
+        last_fetched_offsets = {}
+
+    if offset.isdigit():
+        offset = int(offset)
+
+    topic_partitions = []
+    for partition in partitions:
+        specific_offset = last_fetched_offsets.get(partition, offset) if partition in last_fetched_offsets else offset
+        demisto.debug(f'Getting last offset for partition {partition}, specific offset is {specific_offset}\n')
+        if isinstance(specific_offset, int):
+            specific_offset += 1
```

Comment:
didnt we say this is problematic ?

Reply:
This is the same as Kafka v2, for fetching the first offset the user should input 'earliest'